python/tchannel/tornado/connection.py

```diff
@@ -190,50 +190,49 @@ def _loop(self):
             if context.message.message_type in self.CALL_TYPES:
                 self._messages.put(context)
                 continue
+
             elif context.message_id in self._outstanding:
-                message = self._response_message_factory.build(
+                response = self.response_message_factory.build(
                     context.message_id, context.message
                 )
+                if (context.message.message_type == Types.ERROR or
+                        context.message.flags == FlagsType.none):
```

Comment:
Instead of checking for just error messages, assume all messages are unfragmented except call req/res and their continuations.